src/io/metadata.cpp

```diff
@@ -251,6 +261,25 @@ void Metadata::CheckOrPartition(data_size_t num_all_data, const std::vector<data
         old_weights.clear();
       }
     }
+    // check positions
+    if (position_load_from_file_) {
+      if (positions_.size() > 0 && num_positions_ != num_all_data) {
+        positions_.clear();
+        num_positions_ = 0;
+        Log::Fatal("Positions size doesn't match data size");
```

Comment:
```suggestion
       Log::Fatal("Positions size (%i) doesn't match data size (%i)", num_positions_, num_data_);
```

Since you already have access to them, please print the sizes here, to help users who encounter this error debug what went wrong.